hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/table/manager/HoodieTableManagerClient.java

```diff
@@ -0,0 +1,181 @@
+        new String[] {dbName, tableName, instantRange, config.getDeployUsername(), config.getDeployQueue(),
+            config.getDeployResource(), String.valueOf(config.getDeployParallelism()), config.getDeployExtraParams()});
+    queryParameters.forEach(builder::addParameter);
+
+    String url = builder.toString();
+    LOG.info("Sending request to table management service : (" + url + ")");
+    Response response;
+    int timeout = this.config.getConnectionTimeout() * 1000; // msec
+    int requestRetryLimit = config.getConnectionRetryLimit();
+    int retry = 0;
+
+    while (retry < requestRetryLimit) {
+      try {
+        response = Request.Get(url).connectTimeout(timeout).socketTimeout(timeout).execute();
+        return response.returnContent().asString();
+      } catch (IOException e) {
+        retry++;
+        LOG.warn(String.format("Failed request to server %s, will retry for %d times", url, requestRetryLimit - retry), e);
+        if (requestRetryLimit == retry) {
+          throw e;
+        }
+      }
+
+      try {
+        TimeUnit.SECONDS.sleep(config.getConnectionRetryDelay());
+      } catch (InterruptedException e) {
+        // ignore
+      }
+    }
+
+    throw new IOException(String.format("Failed request to table management service %s after retry %d times", url, requestRetryLimit));
+  }
+
+  private Map<String, String> getParamsWithAdditionalParams(String[] paramNames, String[] paramVals) {
+    Map<String, String> paramsMap = new HashMap<>();
+    paramsMap.put(BASEPATH_PARAM, basePath);
+    ValidationUtils.checkArgument(paramNames.length == paramVals.length);
+    for (int i = 0; i < paramNames.length; i++) {
+      paramsMap.put(paramNames[i], paramVals[i]);
+    }
+    return paramsMap;
+  }
+
+  public void submitCompaction(List<String> instantsToSubmit) {
+    try {
+      executeRequest(SUBMIT_COMPACTION, StringUtils.join(instantsToSubmit.toArray(new String[0]), ","));
+    } catch (Exception e) {
+      LOG.error("Failed to schedule compaction to service", e);
+      failTimes++;
+      checkTolerableSchedule(e);
+      return;
+    }
+
+    failTimes = 0;
+  }
+
+  public void submitClean(List<String> instantsToSubmit) {
+    try {
+      executeRequest(SUBMIT_CLEAN, StringUtils.join(instantsToSubmit.toArray(new String[0]), ","));
+    } catch (Exception e) {
+      LOG.error("Failed to schedule clean to service", e);
+      failTimes++;
+      checkTolerableSchedule(e);
+      return;
+    }
+
+    failTimes = 0;
+  }
+
+  public void submitClustering(List<String> instantsToSubmit) {
+    try {
+      executeRequest(SUBMIT_CLUSTERING, StringUtils.join(instantsToSubmit.toArray(new String[0]), ","));
+    } catch (Exception e) {
+      LOG.error("Failed to schedule clustering to service", e);
+      failTimes++;
+      checkTolerableSchedule(e);
+      return;
+    }
+
+    failTimes = 0;
+  }
+
+  private void checkTolerableSchedule(Exception e) {
```

Comment:
There is already a retry process for request failure, checkTolerableSchedule is just a process to prevent all tasks from failing when table manager service in case have an exception.